lib/smooth/include/smooth/core/network/Wifi.h

```diff
@@ -73,6 +73,8 @@ namespace smooth::core::network
                                             void* event_data);
 
             [[nodiscard]] std::string get_mac_address() const;
+            esp_err_t get_local_mac_address(uint8_t m[6]) const;
```

Comment:
Lets return a bool instead, we succeed or we don't. Unless you actually need know the underlying reason?

I try to keep IDF-types out of the API if possible.

Reply:
o.k.

Reply:
Depending on what error values are originally returned, you could add some error logging if it fails.